site/sfguides/src/getting_started_with_traces/getting_started_with_traces.md

```diff
@@ -0,0 +1,445 @@
+## Creating the Notebook
+Duration: 2
+
+Login to Snowsight using your credentials in Snowflake.
+
+You'll need a Database, Schema, and Warehouse to get started with the notebook.
+
+Run the following sql (in a worksheet) to create the warehouse, database, and schema:
+
+```sql
+CREATE OR REPLACE WAREHOUSE TRACING_QUICKSTART_WH WAREHOUSE_SIZE=XSMALL, INITIALLY_SUSPENDED=TRUE;
+CREATE OR REPLACE DATABASE TRACING_QUICKSTART;
+CREATE OR REPLACE SCHEMA DATA;
+
+```
+
+Refresh Quicksight so the new warehouse will be available.
+
+Go to Projects > Notebooks in the left navigation. Click the + Notebook to create a new Notebook. Name the Notebook `TRACING_QUICKSTART` and Select the `TRACING_QUICKSTART` database and the `DATA` schema you created. Select the `TRACING_QUICKSTART_WH` that was just created, leave the default Run on warehouse selected and click Create.
+
+<!-- ------------------------ -->
+## Setup
+Duration: 2
+
+The first notebook cell will be used to setup the notebook. These variables will be used in subsequent cells using the {{variable_name}} syntax in SQL cells, and by using the variable names in Python cells.
+
+Paste this code into the first cell. If you'd like to pull data for other cities, edit the city_list.
+
+```python
+import streamlit as st
+
+from snowflake.snowpark.context import get_active_session
+session = get_active_session()
+
+user_name = session.sql("SELECT current_user()").collect()[0][0]
+schema_name = "DATA"
+database_name = "TRACING_QUICKSTART"
+
+city_list = [(37.5533204, -122.3059259, 'San Mateo'), (52.3368551, 4.8694973, 'Amsterdam'),(52.5100227, 13.3776724, 'Berlin'), (52.2299305,20.9764142, 'Warsaw'), (18.5645333,73.911966, 'Pune')]
+
+api_key = st.text_input("Enter API key for openweathermap")
+if api_key == "":
+    raise Exception("Configuration needed")
+
+```
+
+When running this cell, it will prompt for entering an API Key. This was done with the streamlit text_input widget. 
+
+To get a free API key, sign up for an account on [OpenWeather](https://home.openweathermap.org/users/sign_up).
+
+After signing up, add a new API key by going to [API keys](https://home.openweathermap.org/api_keys). Create a new key with name `snowflake_key` and click Generate, copy the api key to the clipboard.
+
+After generating the key, go back to the Snowsight notebook and paste the key generated in the dialog and hit enter.
+
+<!-- ------------------------ -->
+## Confgure External API Access
+Duration: 2
+
+In the second notebook cell, a network rule, secret, and external access integration will be configured to allow outbound connectivity to the OpenWeather API. This is necessary because the Python Notebook does not have network access without this configuration. A stage is also created to allow storage of the permanent UDF and procedure. A Permanent (will be stored in the database) UDF and procedure was chosen so these utilities could be called outside the notebook in the future if desired.
```

Comment:
```suggestion
In the second notebook cell you will configure a network rule, secret, and external access integration to allow outbound connectivity to the OpenWeather API. This is necessary because the Python Notebook is secure by default and does not allow network access. A stage is also created to store the permanent UDF and procedure. "Permanent" means that the UDF and stored procedure will be stored in the database so that they can be called from outside the Notebook in the future, such as from a SQL query.
```